operational_analysis/methods/plant_analysis.py

```diff
@@ -672,38 +800,103 @@ def run_AEP_monte_carlo(self):
 
         num_sim = self.num_sim
 
+        # Initialize arrays to store metrics and results
+        self._mc_num_points = np.empty(num_sim, dtype=np.float64)
+        self._r2_score = np.empty(num_sim, dtype=np.float64)
+        self._mse_score = np.empty(num_sim, dtype=np.float64)
+        
+        num_vars = 1
+        if self.reg_winddirection == 'Y' :
+            num_vars = num_vars + 2
+        if self.reg_temperature == 'Y':
+            num_vars = num_vars + 1
```

Comment:
To be really picky - you should be able to compute num_vars as a dimension (using the .shape() method) from some data structure elsewhere in this code. It should not be necessary to keep track of num_vars explicitly.

Reply:
I have thought about this, but couldn't find an easy way to fix this, mainly because num_vars is needed to define `self._mc_slope = np.empty([num_sim,num_vars], dtype=np.float64)` a couple lines after that...

Reply:
Ok. I don't love it, but I can live with this for now. Most of the workarounds probably reduce clarity even moreso than this does.